App/Sagas/Migration.ts

```diff
@@ -32,6 +44,71 @@ export function * migrate(dispatch: Dispatch) {
   yield call(letScreenSleep)
 }
 
+// Can be run on each node online
+export function * runRecurringMigrationTasks () {
+  const announcement = yield select(getAnnouncement)
+  if (announcement) {
+    const {peerId, address, username, previous} = announcement
+    try {
+      yield call(announceMigration, peerId, address, username, previous)
+    } finally {
+      // If no error, mark as successful
+      yield put(MigrationActions.announceSuccess())
```

Comment:
Similar problem here. You're going to mark it as successful no matter what. This action should only get dispatched if we have a successful network request. You can just put this right after the `call(accounceMigration...)`. If that `call` fails it will throw an error and the `MigrationActions.announceSuccess` won't get dispatched and we'll try again in the future.

Reply:
ah yeah, duh